python312Packages.wolframalpha: init at 5.1.3

Duckle29 opened the pull request:
## Description of changes

Added a package for the wolframalpha python module (dependency of sopel-wolfram plugin that I'm also working on)

I've spent a few hours trying to get nixpkgs-review working, but have had no luck.

Now I'll try and do a PR here and see if I can get the PR functionality of it working, or maybe get some hints from the CI here

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Duckle29 committed:
python3Packages.wolframalpha: init at 5.1.3

Co-authored-by: Arne Keller <2012gdwu+github@posteo.de>

Duckle29 commented:
I tested, with `nixpkgs-review`, that I could import the built python libs

FliegendeWurst commented on pkgs/development/python-modules/wolframalpha/default.nix:
```diff
@@ -0,0 +1,50 @@
+buildPythonPackage rec {
+  pname = "wolframalpha";
+  version = "5.1.3";
+  format = "pyproject";
```
```suggestion
  pyproject = true;
```

FliegendeWurst commented on pkgs/development/python-modules/wolframalpha/default.nix:
```diff
@@ -0,0 +1,50 @@
+    changelog = "https://github.com/jaraco/wolframalpha/releases/tag/v${version}";
+    homepage = "https://github.com/jaraco/wolframalpha";
+    description = "Python Client built against the Wolfram|Alpha v2.0 API.";
+    license = lib.licenses.mit;
```
New packages require a maintainer

FliegendeWurst commented on pkgs/development/python-modules/wolframalpha/default.nix:
```diff
@@ -0,0 +1,50 @@
+  ];
+
+  doCheck = false;
+
```
```suggestion
  pythonImportsCheck = [ "wolframalpha" ];
```

FliegendeWurst commented on pkgs/development/python-modules/wolframalpha/default.nix:
```diff
@@ -0,0 +1,50 @@
+  meta = {
+    changelog = "https://github.com/jaraco/wolframalpha/releases/tag/v${version}";
+    homepage = "https://github.com/jaraco/wolframalpha";
+    description = "Python Client built against the Wolfram|Alpha v2.0 API.";
```
```suggestion
    description = "Python Client for the Wolfram|Alpha v2.0 API";
```

Duckle29 committed:
maintainers: add duckle29

Duckle29 commented:
@FliegendeWurst Thanks for the review! 😄 

I've applied your recommendations, and added myself as maintainer for this (I think)
I've also squashed all commits to hopefully have a commit compliant with contributing.md

FliegendeWurst commented on pkgs/development/python-modules/wolframalpha/default.nix:
```diff
@@ -0,0 +1,51 @@
+    multidict
+  ];
+
+  doCheck = false;
```
```suggestion
  doCheck = false; # require network access
```
I'm assuming

FliegendeWurst commented on pkgs/top-level/python-packages.nix:
```diff
@@ -31,6 +31,8 @@ self: super: with self; {
 
   abjad = callPackage ../development/python-modules/abjad { };
 
+  wolframalpha = callPackage ../development/python-modules/wolframalpha { };
```
Would be best to have it sorted :)

mjeplin commented:
Hmm is there a way to re-run the offborg step? seems to have timed out
